.pre-commit-config.yaml

```diff
@@ -1,6 +1,6 @@
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: "v0.1.6"
+    rev: "v0.3.4"
```

Comment:
Since `ruff` is included as a Poetry dev group dependency, would it make sense to include a [repository local hook](https://pre-commit.com/#repository-local-hooks) so as to align versions used both during development as well as in the pre-commit hook? I can see why not too, so feel free to dismiss.

Reply:
This suggestion seems good to me, but I don't want to delay this PR while I learn how to put the repo-local-hooks together. Created new issue to track: https://github.com/cytomining/pycytominer/issues/390